src/coreclr/tools/aot/crossgen2/Program.cs

```diff
@@ -178,6 +179,14 @@ private void ProcessCommandLine(string[] args)
 
         }
 
+        private string GetCompilerVersion()
+        {
+            return  Assembly
+                   .GetExecutingAssembly()
+                   .GetCustomAttribute<AssemblyFileVersionAttribute>()
```

Comment:
```suggestion
                   .GetCustomAttribute<AssemblyInformationalVersionAttribute>()
```
The other tools are standardizing on printing `AssemblyInformationalVersionAttribute` for the `--version` option.

https://github.com/dotnet/command-line-api/blob/d86f87aa98d6392f99a407930e2b6451f6a33372/src/System.CommandLine/Builder/CommandLineBuilderExtensions.cs#L28

https://github.com/dotnet/command-line-api/blob/d86f87aa98d6392f99a407930e2b6451f6a33372/src/System.CommandLine/Builder/CommandLineBuilderExtensions.cs#L620

Do we want to do the same?

Reply:
That will help the potential transition to System.CommandLine. (I have wip changes for it which I've stashed for 67232 testing, I'll get back to it soon'ish)

Reply:
A good observation. I tested both options in Crossgen2, and they yield different results.

```text
AssemblyFileVersionAttribute: 42.42.42.42424
AssemblyInformationalVersionAttribute: 7.0.0-dev
```

The first one shows the compiler version, and the second one shows the .NET version. The question here is which piece of information is more relevant to display directly from _Crossgen2.exe_, or whether we want to display both.

What are your thoughts @trylek?

Reply:
Please build with `./build.sh -p:OfficialBuildId=$(date +%Y%m%d)-99` to get the non-default (42.42.42.42424) versions.